docs/guides/quickfort-alias-guide.rst

```diff
@@ -0,0 +1,850 @@
+into the DF UI.
+
+You use an alias by typing its name into a ``#query`` blueprint cell where you
+want it to be applied. You can use an alias by itself or as part of a larger
+sequence, potentially with other aliases. If the alias is the only text in the
+cell, the alias name is matched and its expansion is used. If the alias has
+other keys before or after it, the alias name must be surrounded in curly
+brackets (:kbd:`{` and :kbd:`}`). An alias can be surrounded in curly brackets
+even if it is the only text in the cell, it just isn't necesary. For example,
+the following blueprint uses the ``aliasname`` alias by itself in the first
+two rows and uses it as part of a longer sequence in the third row:
+
+::
+
+    #query apply alias 'aliasname' in three different ways
+    aliasname
+    {aliasname}
+    literaltext{aliasname}literaltext
+
+For a more concrete example of an alias definition, a simple alias that
+configures a stockpile to have no bins (:kbd:`C`) and no barrels (:kbd:`E`)
+assigned to it would look like this:
+
+::
+
+    nocontainers: CE
+
+The alias definition can also contain references to other aliases by including
+the alias names in curly brackets. For example, ``nocontainers`` could be
+equivalently defined like this:
+
+::
+
+    nobins: C
+    nobarrels: E
+    nocontainers: {nobins}{nobarrels}
+
+Aliases used in alias definitions *must* be surrounded by curly brackets, even
+if they are the only text in the definition:
+
+::
+
+    alias1: text1
+    alias2: alias1
+    alias3: {alias1}
+
+Here, ``alias1`` and ``alias3`` expand to ``text1``, but ``alias2`` expands to
+the literal text ``alias1``.
+
+Keycodes
+~~~~~~~~
+
+Non-printable characters, like the arrow keys, are represented by their
+keycode name and are also surrounded by curly brackets, like ``{Right}`` or
+``{Enter}``. Keycodes are used exactly like aliases -- they just have special
+expansions that you wouldn't be able to write yourself. In order to avoid
+naming conflicts between aliases and keycodes, the convention is to start
+aliases with a lowercase letter.
+
+Any keycode name from the DF interface definition file
+(data/init/interface.txt) is valid, but only a few keycodes are actually
+useful for blueprints:
+
+::
+
+    Up
+    Down
+    Left
+    Right
+    Enter
+    ESC
+    Backspace
+    Space
+    Tab
+
+Repetitions
+~~~~~~~~~~~
+
+Anything enclosed within curly brackets can also have a number, indicating how
+many times that alias or keycode should be repeated. For example:
+``{togglesequence 9}`` or ``{Down 5}`` will repeat the ``togglesequence``
+alias nine times and the ``Down`` keycode five times, respectively.
+
+Modifier keys
+~~~~~~~~~~~~~
+
+Ctrl, Alt, and Shift modifiers can be specified for the next key by adding
+them into the key sequence. For example, Alt-h is written as ``{Alt}h``.
+
+Note that DF does not handle keys that have more than a single modifier, so
+sequences like ``{Ctrl}{Alt}a`` will result in an error.
```

Comment:
This isn't quite true - the `interface_key`s starting with `CUSTOM_` are only defined for up to one modifier, but DF supports binding any interface_key to combinations with multiple modifiers (e.g. `ctrl-alt-e` gets saved as `[SYM:6:e]`, where `6 == 4 | 2 == CTRL | ALT`).

Perhaps "DF" should be replaced with "quickfort" here? In any case, the default interface.txt (well, mine, which has minimal modifications) only uses `1`, `2`, and `4` modifiers for `SYM` entries, so this is a relatively unimportant feature to support in my opinion.

Reply:
I removed the note. Thanks! FTR, quickfort code does support the general case of multiple modifiers. I just noticed that interface.txt didn't have any multiple-modifier keys when I tried to test the functionality. I forgot that custom definitions could be added.